traits/has_traits.py

```diff
@@ -3455,6 +3455,55 @@ class HasStrictTraits ( HasTraits ):
     """
     _ = Disallow   # Disallow access to any traits not explicitly defined
 
+#-------------------------------------------------------------------------------
+#  'HasRequiredTraits' class:
+#-------------------------------------------------------------------------------
+
+class HasRequiredTraits(HasStrictTraits):
+    """ This class builds on the functionality of HasStrictTraits and ensures
+     that any object attribute with required=True (metadata) must be passed
+     as an argument on object initialization.
+
+    This can be useful in cases where an object has traits which are required
+    for it to function correctly.
+
+    Raises
+    ------
+    TraitError
+        If a required trait is not passed as an argument
+
+    Examples
+    --------
+    A class with required traits:
+    >>> class RequiredTest(HasRequiredTraits):
+    >>>     required_trait = Any(required=True)
+    >>>     non_required_trait = Any()
+
+    Making an instance of a HasRequiredTraits class:
+    >>> test_instance = RequiredTest(required_trait=13, non_required_trait=11)
+    >>> test_instance2 = RequiredTest(required_trait=13)
+
+    Forgetting to specify a required trait:
+    >>> test_instance = RequiredTest(non_required_trait=11)
+    traits.trait_errors.TraitError: The following required traits were not
+    passed as a keyword argument: required_trait.
+    """
+
+    def __init__(self, **traits):
+        missing_required_traits = []
+        for name in self.trait_names(required=True):
+            if name not in traits:
+                missing_required_traits.append(name)
+        if missing_required_traits:
+            raise TraitError(
+                "The following required traits were not passed as a "
+                "keyword argument: {names}.".format(
```

Comment:
Wording nit: I'd probably drop the "as a keyword argument", here; we're all used to having to pass traits as keyword arguments anyway. How about just: "The following required traits were not provided: ..."?